From 5c06ce596320af7f34fc9fc9f7b4373ba5c7dbe2 Mon Sep 17 00:00:00 2001
From: P Praneesh <quic_ppranees@quicinc.com>
Date: Tue, 30 Apr 2024 10:48:04 +0530
Subject: [PATCH] QSDK: wifi: ath11k: ath12k: use netdev_alloc_skb_fast instead of
 dev_alloc_skb

Currently dev_alloc_skb is used by CE and data path related rings.
Buffers allocated by data path rx goes via fast recycler to avoid clearing
few fields. When non data path rings allocates skb through the recycler,
the resultant skb may contains stale values. Fix it by calling
netdev_alloc_skb_fast instead dev_alloc_skb. netdev_alloc_skb_fast
internally performs skb recycler related functionality whereas
dev_alloc_skb follows traditional skb allocation flow.

Patch-dependency: none

Signed-off-by: P Praneesh <quic_ppranees@quicinc.com>
---
 drivers/net/wireless/ath/ath11k/dp_rx.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/wireless/ath/ath11k/dp_rx.c b/drivers/net/wireless/ath/ath11k/dp_rx.c
index 2c88986..be8bfe0 100644
--- a/drivers/net/wireless/ath/ath11k/dp_rx.c
+++ b/drivers/net/wireless/ath/ath11k/dp_rx.c
@@ -423,8 +423,14 @@ int ath11k_dp_rxbufs_replenish(struct ath11k_base *ab, int mac_id,
 	num_remain = req_entries;
 
 	while (num_remain > 0) {
+#ifdef CPTCFG_MAC80211_SFE_SUPPORT
+		skb = netdev_alloc_skb_fast(NULL,
+					    DP_RX_BUFFER_SIZE +\
+					    DP_RX_BUFFER_ALIGN_SIZE);
+#else
 		skb = dev_alloc_skb(DP_RX_BUFFER_SIZE +
 				    DP_RX_BUFFER_ALIGN_SIZE);
+#endif
 		if (!skb)
 			break;
 
-- 
2.34.1

